setupext.py

```diff
@@ -1211,7 +1131,14 @@ def get_extension(self):
         pkg_config.setup_extension(
             ext, 'libpng', default_libraries=['png', 'z'],
             alt_exec='libpng-config --ldflags')
-        Numpy().add_flags(ext)
+        # We call this twice: once early, to check whether png exists (from
```

Comment:
this hack won't be necessary after #11983 goes in so let's wait for that before rebasing this PR